### DISCUSSION TITLE
Run the simulation with an external time-history aerodynamic force

### DISCUSSION THREAD

**Discussion by Giorgio-C-30** [OpenFAST/openfast#2505] [General]
Hi all,

I would like to know if there is a possibility in Openfast to run a simulation by applying a time-history aerodynamic force at the hub height, instead of computing it with Aerodyn module.
I would like to compute the aerodynamic force through a 3-party software, apply this force at the tower top and evaluate the system response. 

I would appreciate if anybody knows if it is possible or has already done something similar.

Many thanks in advance. Kind regards,
Giorgio

**Comment by jjonkman** [comment]
Dear @Giorgio-C-30,

The structural control (StC) submodel of ServoDyn allows you to specify user-prescribed time series of applied loads (forces, moments) at user-specified locations in the blades, nacelle, tower, and substructure.  So, you can disable AeroDyn, add a nacelle-based StC in ServoDyn, and provide your load time history in the StC input file.  Please note that this approach will not account for relative motion effects, i.e., the aerodynamic loads will not depend on structural response. 

Best regards,

**Comment by Giorgio-C-30** [comment]
Hi Jonkman,

Thank you so much for your response. Can you please point out where I can find a guide how to use the structural control or an example from which I can take inspiration?

Kind regards,
Giorgio

**Reply by jjonkman** [reply]
Hi @Giorgio-C-30,

StC is documented here: https://openfast.readthedocs.io/en/main/source/user/servodyn-stc/StC_index.html.

You can find an example in the r-test: https://github.com/OpenFAST/r-test/tree/main/glue-codes/openfast/StC_test_OC4Semi.

Best regards,
